### PR TITLE
Pointcloud to voxelgrid conversion

### PR DESCRIPTION
**Pull request by JerryJiehanWang**
Pointcloud to voxelgrid conversion

### CODE REVIEW

File: kaolin/ops/conversions/pointcloud.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+__all__ = ['pointclouds_to_voxelgrids']
+
+def pointclouds_to_voxelgrids(pointclouds, resolution, return_sparse=False):
```
Comment (Caenorst) [inline kaolin/ops/conversions/pointcloud.py]: shouldn't we adopt the same interface than `trianglemesh_to_voxelgrids` ? https://github.com/NVIDIAGameWorks/kaolin/blob/master/kaolin/ops/conversions/trianglemesh.py#L20

Comment (JerryJiehanWang) [reply kaolin/ops/conversions/pointcloud.py]: Do you mean we should also add origin and scale as arguments?

Comment (Caenorst) [reply kaolin/ops/conversions/pointcloud.py]: Yes!

Comment (JerryJiehanWang) [reply kaolin/ops/conversions/pointcloud.py]: Added, as well as the tests.

File: kaolin/ops/conversions/pointcloud.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+__all__ = ['pointclouds_to_voxelgrids']
+
+def _points_to_voxelgrids(points, resolution, return_sparse=False):
```
Comment (Caenorst) [inline kaolin/ops/conversions/pointcloud.py]: maybe add the prefix `_base` like for: https://github.com/NVIDIAGameWorks/kaolin/blob/master/kaolin/ops/mesh/trianglemesh.py#L26

Also maybe we can add some details about the behavior? (i.e: For what I understand, it only convert points from 0 to 1)

File: kaolin/ops/conversions/pointcloud.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+# limitations under the License.
+
+import torch
+
+__all__ = ['pointclouds_to_voxelgrids']
+
+def _base_points_to_voxelgrids(points, resolution, return_sparse=False):
+    r"""Converts points to voxelgrids. This is the base function for both trianglemeshes_to_voxelgrids
+    and pointclouds_to_voxelgrids. For point cloud, the points are the original points. For Triangle Mesh,
+    the points are sampled along the edges. The points are normalized into range [0, 1].
```
Comment (Caenorst) [inline kaolin/ops/conversions/pointcloud.py]: I'm not sure we should reference mesh behavior, the behavior that you described here ("For Triangle Meshes, the points are samples along the edges") has nothing to do with this function, isn't it? If you changed the way you sampled points along the edges this function would still be the same the description would be false, which is difficult to maintain.

Also, are the points normalized? It seemed to be that only points in the range [0, 1] (before doing anything to them) were sampled.

Comment (JerryJiehanWang) [reply kaolin/ops/conversions/pointcloud.py]: Yes, the points are normalized. So will it be sufficient to say that the function converts normalized points in the range [0, 1] to voxelgrid?

Comment (Caenorst) [reply kaolin/ops/conversions/pointcloud.py]: My understanding is that the point cloud MAY not be normalized, but regardless of that this base function will ONLY voxelize points that are in the range [0, 1]. Am I correct?

**Review by Caenorst — APPROVED** [review]
LGTM, Thanks!
